File: 268-missing-number/missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let n = nums.length;
    var sum = n*(n+1) / 2;

    for(let i = 0; i <nums.length; i++){
        sum -= nums[i];
    }
    return sum;
};

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        HashSet<Integer> a = new HashSet<Integer>();
        for(int i = 0; i < nums.length; i++){
            a.add(nums[i]);
        }

        for(int i = 0; i<=nums.length; i++){
            if(!a.contains(i)){
                return i;
            }
        }
        return -1;
    }
}